cpp/include/ucxx/delayed_submission.h

```diff
@@ -150,10 +152,17 @@ class BaseDelayedSubmissionCollection {
         item = std::move(_collection.front());
         _collection.pop_front();
         if (_canceled.erase(item.first)) continue;
+        _processing = std::optional<ItemIdType>{item.first};
       }
 
       processItem(item.first, item.second);
     }
```

Comment:
Right, but the loop may continue and we need to acquire `_mutex` to prevent a race with `cancel()`, so I think it should be a reasonable compromise to just do it at the end of the loop https://github.com/rapidsai/ucxx/pull/303/files#diff-be1ae222c4b0723e2817858a72e0815d919a1c20816ffebfa18ff7dc06a4457fR162-R164. WDYT?